src/controllers/bookings.cr

```diff
@@ -217,6 +225,14 @@ class Bookings < Application
     clashing_bookings = check_clashing(existing_booking)
     render :conflict, json: clashing_bookings.first if clashing_bookings.size > 0
 
+    # check concurrent bookings don't exceed booking limits
+    if booking_limits = tenant.booking_limits.as_h?
+      if limit = booking_limits[existing_booking.booking_type]?
+        concurrent_bookings = check_concurrent_bookings(existing_booking)
+        render :conflict, json: concurrent_bookings.first if concurrent_bookings.size >= limit.as_i
+      end
+    end
+
```

Comment:
Repeated code (same as L77-84) so I would throw it into a function

Reply:
I have moved it to a method (#check_booking_limits) that returns the first conflicting booking if any.